### DISCUSSION TITLE
It seems that the current MultiHeadAttention class is only a SingleHeadAttention class.

### DISCUSSION THREAD

**Discussion by yujieNC** [kornia/kornia#2155] [Potential Issue]
It seems that the  current MultiHeadAttention class is only a SingleHeadAttention class.
https://github.com/kornia/kornia/blob/91ee3bb40a21bef82cacc080e6ade081f8075f81/kornia/contrib/vit.py#L58
@edgarriba

**Comment by edgarriba** [comment]
Can you send an improvement for this ?

**Reply by yujieNC** [reply]
OK. I have created a [PR](https://github.com/kornia/kornia/pull/2184) for this discussion.
